### PR TITLE
Move to cmd/runc.

### PR DESCRIPTION
**Pull request by gwoo**
- Make bin/runc 
- Add SpecConfig Interface.

### CODE REVIEW

**Comment by stevvooe** [comment]
@gwoo If you could separate the move to cmd/runc into a separate PR, we can likely get it merged more quickly. The rest of this code will take some time to review.

**Comment by gwoo** [comment]
@stevvooe most of these changes were necessary to make the move to cmd/runc. We have to be able to use runc as a package and that needs to have some abstracted functionality to be used by all the commands. Namely the "Spec".

File: spec.go [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+	NewConfig() (*configs.Config, error)
+	AddNamepsaces(config *configs.Config) error
+	AddMounts(config *configs.Config) error
+	AddDevices(config *configs.Config) error
+	AddUserNamespace(config *configs.Config) error
+	AddGroups(config *configs.Config) error
+	SetReadOnly(config *configs.Config) error
+}
+
+type Spec struct {
```
Comment (stevvooe) [inline spec.go]: Please document the struct and fields.

File: spec.go [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
-	"encoding/json"
-	"fmt"
-	"runtime"
-
-	"github.com/codegangsta/cli"
+	"github.com/opencontainers/runc/libcontainer/configs"
 )
 
-const cpuQuotaMultiplyer = 100000
+type SpecConfig interface {
```
Comment (stevvooe) [inline spec.go]: Please document this interface and its methods.

**Comment by stevvooe** [comment]
@gwoo So, by exporting these symbols, we are making the decision that these don't change.

Let's at least make sure that anything that is exported gets proper documentation.

**Comment by gwoo** [comment]
@stevvooe On it.

**Comment by gwoo** [comment]
@stevvooe Add some docs and rebased on master.

**Comment by crosbymichael** [comment]
I kinda like things being at the root of the repo. It's simple and works, you can go get, go build, easily.  There is no hunting things down and trying to figure out where the code is, it's in one place.

**Comment by stevvooe** [comment]
@crosbymichael Having commands in the root directory goes against the Go convention. It also means that packages cannot be at the root of the repository, leading to extra path components to use dependencies.

**Comment by crosbymichael** [comment]
@stevvooe where is the go convention?  I have not read anything about it.

**Comment by stevvooe** [comment]
Most of the [Go tools](https://godoc.org/golang.org/x/tools) follow this convention. Check out [Dave Cheney's Post](http://dave.cheney.net/2014/12/01/five-suggestions-for-setting-up-a-go-project) on the topic. A large number of open source Go projects also follow this convention.

One of the main benefits is `import "github.com/opencontainers/runc"` does the expected of importing a package named `runc`, rather than a main package.

**Comment by crosbymichael** [comment]
We don't have a package named runc and don't want one

**Pull request closed without merging**